### ISSUE TITLE
Action Text integration tests raises ndefined method `render_to_string' for nil:NilClass

### ISSUE THREAD

**Issue by yahonda** [rails/rails#46113]
Reported at the Rails Discord server https://discord.com/channels/849034466856665118/974005005768069211/1022319249999544340 

### Steps to reproduce

```ruby
git clone https://github.com/rails/rails
cd rails/actiontext
bundle install
bin/test test/integration/
```

### Expected behavior
It should pass.

### Actual behavior
It raises NoMethodError: undefined method `render_to_string' for nil:NilClass

```ruby
$ bin/test test/integration/
Run options: --seed 17477

# Running:

E

Error:
ActionText::MailerRenderTest#test_uses_default_url_options:
NoMethodError: undefined method `render_to_string' for nil:NilClass

        (renderer || default_renderer).render_to_string(*args, &block)
                                      ^^^^^^^^^^^^^^^^^
    /home/yahonda/src/github.com/rails/rails/actiontext/lib/action_text/rendering.rb:26:in `render'
    /home/yahonda/src/github.com/rails/rails/actiontext/lib/action_text/rendering.rb:13:in `render'
    /home/yahonda/src/github.com/rails/rails/actiontext/lib/action_text/content.rb:87:in `to_rendered_html_with_layout'
    /home/yahonda/src/github.com/rails/rails/actiontext/lib/action_text/content.rb:95:in `to_s'
    /home/yahonda/src/github.com/rails/rails/actiontext/lib/action_text/content.rb:57:in `append_attachables'
    /home/yahonda/src/github.com/rails/rails/actiontext/test/integration/mailer_render_test.rb:11:in `block in <class:MailerRenderTest>'


bin/test test/integration/mailer_render_test.rb:6

.....

Finished in 0.325615s, 18.4267 runs/s, 24.5689 assertions/s.
6 runs, 8 assertions, 0 failures, 1 errors, 0 skips
```

### System configuration
**Rails version**: main branch

**Ruby version**: ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-linux]

**Comment by yahonda** [comment]
I have found a workaround for this exception by running with `test/template/form_helper_test.rb`. I do not know why yet.

```
$ bin/test test/integration/ test/template/form_helper_test.rb
Run options: --seed 62742

# Running:

.................

Finished in 0.312267s, 54.4405 runs/s, 76.8572 assertions/s.
17 runs, 24 assertions, 0 failures, 0 errors, 0 skips
$
```

**Comment by clouvet** [comment]
@yahonda, I was able to reproduce this locally by running `bin/test test/integration/controller_render_test.rb` with some specific seeds. Seed `18309` resulted in no failures, while seed `49319` resulted in failures for all three tests that called `ActionText::Content.new.append_attachables`. 

After a little more inspection, I determined that when the failures occurred, both `render` and `default_renderer` were nil in the `ActionText::Rendering` module's `render` method. 

`append_attachables` calls the  `render` method, but as those three failing tests are currently written there's no invocation of a controller preceding the call to `append_attachables`, so there's no chance for the `renderer` or `default_renderer` variables to be set. 

To confirm that's the case, I can add `controller = MessagesController` to those tests, and all three pass, even with the bad seed. That also works on `bin/test test/integration/mailer_render_test.rb --seed 17477`. 

So tests that call `render` will pass if they are preceded by tests that set up the controller and the previous renderer, which accounts for the intermittent failures. That also explains why running the whole suite is fine.  

To address this problem, the default renderer could be set in `test_helper`:
```ruby
ActionText::Content.default_renderer = Class.new(ActionController::Base).renderer
```

Or the default renderer could be set in the `render` method if it doesn't already exist:
```ruby
def render(*args, &block)
  default_renderer ||= Class.new(ActionController::Base).renderer
  (renderer || default_renderer).render_to_string(*args, &block)
end
```

I don't know if either of those proposals are ideal solutions, but I'm happy to make a pull request.
